crates/interpreter/src/instructions/host_env.rs

```diff
@@ -33,24 +31,24 @@ pub fn difficulty<H: Host, SPEC: Spec>(interpreter: &mut Interpreter, host: &mut
     }
 }
 
-pub fn gaslimit(interpreter: &mut Interpreter, host: &mut dyn Host) {
+pub(super) fn gaslimit(interpreter: &mut Interpreter, host: &mut dyn Host) {
     gas!(interpreter, gas::BASE);
     push!(interpreter, host.env().block.gas_limit);
 }
 
-pub fn gasprice(interpreter: &mut Interpreter, host: &mut dyn Host) {
+pub(super) fn gasprice(interpreter: &mut Interpreter, host: &mut dyn Host) {
     gas!(interpreter, gas::BASE);
     push!(interpreter, host.env().effective_gas_price());
 }
 
-pub fn basefee<SPEC: Spec>(interpreter: &mut Interpreter, host: &mut dyn Host) {
+// EIP-3198: BASEFEE opcode
+pub(super) fn basefee<SPEC: Spec>(interpreter: &mut Interpreter, host: &mut dyn Host) {
+    check!(interpreter, LONDON);
     gas!(interpreter, gas::BASE);
-    // EIP-3198: BASEFEE opcode
-    check!(interpreter, SPEC::enabled(LONDON));
```

Comment:
I feel like the spec check should be before any gas checks, is this right @rakita?

Reply:
It is not a bug as both actions would result in the same behaviour (REVERT), but it is a lot nicer and more consistent to  do `check` first then `gas`